### PR TITLE
[style] add cmake files check and format

### PR DESCRIPTION
**Pull request by bukepo**

### CODE REVIEW

File: examples/apps/cli/CMakeLists.txt [hunk 1]
```diff
@@ -26,41 +26,23 @@
...
 
-target_link_libraries(ot-cli-ftd PRIVATE
-    openthread-cli-ftd
-    ${OT_PLATFORM_LIB}
-    openthread-ftd
-    ${OT_PLATFORM_LIB}
-    mbedcrypto
-    ot-config
+target_link_libraries(
+    ot-cli-ftd PRIVATE openthread-cli-ftd ${OT_PLATFORM_LIB} openthread-ftd ${OT_PLATFORM_LIB} mbedcrypto ot-config
```
Comment (wgtdkp) [inline examples/apps/cli/CMakeLists.txt:41]: I think the original format is more readable. can we customize the configuration file of cmake-format to use the original format?

File: .cmake-format.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+dangle_parens: true
+first_comment_is_literal: true
+keyword_case: 'upper'
+line_width: 120
+tab_size: 4
```
Comment (wgtdkp) [inline .cmake-format.yaml:5]: How about customize this configuration file to minimize changes to existing cmake files?

File: CMakeLists.txt [hunk 3]
```diff
@@ -63,24 +66,33 @@ if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_C_COMPILER_ID MATCHES "(Apple)?[Cc]lang")
...
-    OUTPUT_VARIABLE OT_VERSION OUTPUT_STRIP_TRAILING_WHITESPACE
+    OUTPUT_VARIABLE OT_VERSION
+    OUTPUT_STRIP_TRAILING_WHITESPACE
 )
 message(STATUS "Version: ${OT_VERSION}")
 
-target_compile_definitions(ot-config INTERFACE
-    "PACKAGE_NAME=\"OPENTHREAD\""
-    "PACKAGE_VERSION=\"${OT_VERSION}\""
-)
```
Comment (wgtdkp) [inline CMakeLists.txt:81]: What about enforcing style of this kind of statement to like:
```cmake
target_compile_definitions(ot-config
    INTERFACE
        "PACKAGE_NAME=\"OPENTHREAD\""
        "PACKAGE_VERSION=\"${OT_VERSION}\""
)
```
?

File: examples/apps/cli/CMakeLists.txt [hunk 4]
```diff
@@ -26,41 +26,35 @@
...
-    ot-config
+target_link_libraries(
+    ot-cli-ftd
+    PRIVATE openthread-cli-ftd
+            ${OT_PLATFORM_LIB}
+            openthread-ftd
+            ${OT_PLATFORM_LIB}
+            mbedcrypto
+            ot-config
 )
```
Comment (wgtdkp) [inline examples/apps/cli/CMakeLists.txt:48]:
```suggestion
target_link_libraries(ot-cli-ftd
    PRIVATE
        openthread-cli-ftd
        ${OT_PLATFORM_LIB}
        openthread-ftd
        ${OT_PLATFORM_LIB}
        mbedcrypto
        ot-config
)
```

File: examples/platforms/cc2652/CMakeLists.txt [hunk 5]
```diff
@@ -66,33 +76,19 @@ add_library(openthread-cc2652
...
-target_link_libraries(openthread-cc2652
-    PUBLIC
-        cc13x2-cc26x2-driver
-    PRIVATE
-        ${OT_MBEDTLS}
-        ot-config
+target_link_libraries(
+    openthread-cc2652
+    PUBLIC cc13x2-cc26x2-driver
+    PRIVATE ${OT_MBEDTLS} ot-config
```
Comment (wgtdkp) [inline examples/platforms/cc2652/CMakeLists.txt:82]: I think the old style is better.

File: CMakeLists.txt (hunk 3 above)
Comment (bukepo) [reply CMakeLists.txt:81]: I don't think the tool support it. If you know how to make it, I'm glad to adopt this style.

File: examples/platforms/cc2652/CMakeLists.txt (hunk 5 above)
Comment (bukepo) [reply examples/platforms/cc2652/CMakeLists.txt:82]: Yes, I agree. But I think the new style is also acceptable. I don't think there will be a perfect tool for us to format code. This style just make sure the style is consistent. I think a consistency style is more important.

File: script/make-pretty [hunk 6]
```diff
@@ -67,6 +67,28 @@ readonly OT_CLANG_SOURCES=('*.c' '*.cc' '*.cpp' '*.h' '*.hpp')
...
+    echo -e '====================='
+
+    git ls-files "${OT_CMAKE_FILES[@]}" | grep -v -E "^($(echo "${OT_EXCLUDE_DIRS[@]}" | tr ' ' '|'))" \
+        | xargs -n3 -P"$OT_BUILD_JOBS" cmake-format -i
+}
+
+do_cmake_check()
+{
+    echo -e '====================='
+    echo -e '     check cmake    '
```
Comment (Irving-cl) [inline script/make-pretty:85]:
```suggestion
    echo -e '     check cmake     '
```
align

**Comment by bukepo** [comment]
close for now.

**Pull request closed without merging**